src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -135,6 +155,58 @@ public override void ExecuteCmdlet()
                 {
                     var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.GetProfile<AzureRmProfile>());
 
+                    if (this.ParameterSetName.Equals(MetadataParameterSet, StringComparison.Ordinal))
+            		{
+                        // Simply use built-in environments if the ARM endpoint matches the ARM endpoint for a built-in environment
+                        ResourceManagerEndpoint = ARMEndpoint;
+
+            		    var publicEnvironment =
+            		        AzureEnvironment.PublicEnvironments.FirstOrDefault(
+            		                                                           env =>
+            		                                                           AzureRmProfileProvider.Instance.Profile.GetEnvironment(env.Key)
```

Comment:
There is no reason to use the GetEnvironment call, you should just be able to use env.Value to get this data